src/integration_tests/deploy_prism/definition.ts

```diff
@@ -1,100 +1,324 @@
-): Promise<PrismMarketInfo> {
-    let cw20_code_id = await Cw20CodeId(lcd_client, sender);
-    console.log(`=======================`);
+	let yluna_token_addr = await init_token(lcd_client, sender, cw20_code_id, yluna_token_config);
+	return yluna_token_addr
+}
 
-    // source: 
-    // https://finder.terra.money/mainnet/address/terra1dh9478k2qvqhqeajhn75a2a7dsnf74y5ukregw
-    let prism_token_config = {
+async function init_prism(lcd_client: LCDClient, sender: Wallet, cw20_code_id: number) {
+	// instantiate prism token
+	// source: 
+	// https://finder.terra.money/mainnet/address/terra1dh9478k2qvqhqeajhn75a2a7dsnf74y5ukregw
+	let prism_token_config = {
 		name: "Prism governance token",
 		symbol: "PRISM",
 		decimals: 6,
-		initial_balances: [],
+		initial_balances: [
+			{
+				address: sender.key.accAddress,
+				amount: "10000000000000000"
+			}
+		],
 		mint: {
 			minter: sender.key.accAddress,
 		},
 	};
 
 	let prism_token_addr = await init_token(lcd_client, sender, cw20_code_id, prism_token_config);
+	return prism_token_addr
+}
+
+async function init_prism_governance(
+	lcd_client: LCDClient,
+	sender: Wallet,
+	cw20_code_id: number,
+	prism_token_addr: Addr
+): Promise<PrismGovernanceInfo> {
+	let prism_gov_code_id = await store_contract(lcd_client, sender, prism_gov_wasm)
+	console.log(`prism_gov uploaded\n\tcode_id: ${prism_gov_code_id}`);
+
+	let prism_gov_config = PrismGovConfig(prism_token_addr, cw20_code_id);
+	let init_contract_res = await instantiate_contract_raw(lcd_client, sender, sender.key.accAddress, prism_gov_code_id, prism_gov_config);
+
+	let prism_gov_deployment_addr = getContractAddress(init_contract_res);
+
+	var xprism_token_addr = ''
+	let contract_events = getContractEvents(init_contract_res);
+	for (let contract_event of contract_events) {
+		let xprism_token_addr_from_contract = contract_event["xprism_token_addr"];
+		if (xprism_token_addr_from_contract !== undefined) {
+			xprism_token_addr = xprism_token_addr_from_contract;
+		}
+	}
+
+	return {
+		prism_gov_deployment_addr,
+		xprism_token_addr,
+		prism_gov_config
+	}
+}
+
+async function init_prism_launch_pool(
+	lcd_client: LCDClient,
+	sender: Wallet,
+	prism_token_addr: Addr,
+	yluna_token_addr: Addr,
+	xprism_token_addr: Addr,
+	prism_gov_deployment_addr: Addr,
+): Promise<Addr> {
+	let prism_launch_pool_code_id = await store_contract(lcd_client, sender, prism_launch_pool_wasm)
+	console.log(`prism_launch_pool uploaded\n\tcode_id: ${prism_launch_pool_code_id}`);
+
+	let prism_launch_pool_config = PrismLaunchPoolConfig(
+		sender.key.accAddress,
+		sender.key.accAddress,
+		yluna_token_addr,
+		prism_token_addr,
+		xprism_token_addr,
+		prism_gov_deployment_addr
+	);
+	let prism_launch_pool_address = await instantiate_contract(lcd_client, sender, sender.key.accAddress, prism_launch_pool_code_id, prism_launch_pool_config);
+
+	return prism_launch_pool_address
+}
+
+// TODO: may not be needed so leaving to the end for the nex-prism-convex work
```

Comment:
think it is finished, isnt it?

Reply:
i left out completing this one since the nex-prism-vault instantiation didn't need it. i'll remove it after adding a few more integration tests to make sure its not used, or if it is I'll finish it up